mcp/sensors/star_camera_receive.c

```diff
@@ -0,0 +1,347 @@
+    SET_SCALED_VALUE(sc2_make_static_hp_mask_Addr, scr.makeHP);
+    SET_SCALED_VALUE(sc2_use_static_hp_mask_Addr, scr.useHP);
+    SET_SCALED_VALUE(sc2_time_limit_Addr, scr.solveTimeLimit);
+    SET_SCALED_VALUE(sc2_logodds_Addr, scr.logOdds);
+    SET_SCALED_VALUE(sc2_latitude_Addr, scr.latitude);
+    SET_SCALED_VALUE(sc2_longitude_Addr, scr.longitude);
+    SET_SCALED_VALUE(sc2_hm_Addr, scr.heightWGS84);
+    SET_SCALED_VALUE(sc2_focus_position_Addr, scr.focusPos);
+    SET_SCALED_VALUE(sc2_focus_inf_Addr, scr.setFocusInf);
+    SET_SCALED_VALUE(sc2_aperture_steps_Addr, scr.apertureSteps);
+    SET_SCALED_VALUE(sc2_max_aperture_Addr, scr.maxAperture);
+    SET_SCALED_VALUE(sc2_min_focus_pos_Addr, scr.minFocusPos);
+    SET_SCALED_VALUE(sc2_max_focus_pos_Addr, scr.maxFocusPos);
+    SET_SCALED_VALUE(sc2_current_aperture_Addr, scr.aperture);
+    SET_SCALED_VALUE(sc2_exposure_time_Addr, scr.exposureTime);
+    SET_SCALED_VALUE(sc2_focus_mode_Addr, scr.focusMode);
+    SET_SCALED_VALUE(sc2_start_focus_pos_Addr, scr.startPos);
+    SET_SCALED_VALUE(sc2_end_focus_pos_Addr, scr.endPos);
+    SET_SCALED_VALUE(sc2_focus_step_Addr, scr.focusStep);
+    SET_SCALED_VALUE(sc2_photos_per_focus_Addr, scr.photosPerStep);
+}
+
+static void where_to_unpack(struct star_cam_return data, int which_sc) {
+    if (which_sc == 1) {
+        assign_param_data_to_channel_sc1(data);
+        return;
+    } else if (which_sc == 2) {
+        assign_param_data_to_channel_sc2(data);
+        return;
+    } else {
+        blast_err("Invalid star camera address provided to decider");
+    }
+}
+
+// points the listening thread at the right status boolean to watch
+static int check_sc_thread_bool(int which) {
+    if (which == 1) {
+        return CommandData.sc_bools.sc1_param_bool;
+    } else if (which == 2) {
+        return CommandData.sc_bools.sc2_param_bool;
+    } else {
+        blast_err("Invalid star camera to check thread bool %d", which);
+        return -1;
+    }
+}
+
+static int check_for_reset(int which) {
+    if (which == 1 && CommandData.sc_resets.reset_sc1_param) {
+        // reset the flag
+        CommandData.sc_resets.reset_sc1_param = 0;
+        return 1;
+    } else if (which == 2 && CommandData.sc_resets.reset_sc2_param) {
+        CommandData.sc_resets.reset_sc2_param = 0;
+        return 1;
+    } else {
+        return 0;
+    }
+}
+
+// Now that we have the general boilerplate and useful functions out of the way
+// I will add the thread function that
+
+void *parameter_receive_thread(void *args) {
+    struct socket_data * socket_target = args;
+    int first_time = 1;
+    int sockfd;
+    struct addrinfo hints, *servinfo, *servinfoCheck;
+    int returnval;
+    int numbytes;
+    int listening = 1;
+    int which_sc;
+    struct sockaddr_storage their_addr;
+    struct star_cam_return received_parameters;
+    socklen_t addr_len;
+    char ipAddr[INET_ADDRSTRLEN];
+    memset(&hints, 0, sizeof hints);
```

Comment:
`sizeof` is a unary oeprator, so it can be used in this way if the operand isn't a type. I'm guessing this may be a Beej-ism, so maybe not really your fault, however, I think it's better style to always include () with sizeof().

Reply:
100% a beej-ism. please do not question my all-mighty UDP teacher. (but i'll fix it)